client/definitions.go

```diff
@@ -25,35 +25,79 @@ type CollectionDefinition struct {
 // GetFieldByName returns the field for the given field name. If such a field is found it
 // will return it and true, if it is not found it will return false.
 func (def CollectionDefinition) GetFieldByName(fieldName string) (FieldDefinition, bool) {
-	collectionField, ok := def.Description.GetFieldByName(fieldName)
-	if ok {
-		schemaField, ok := def.Schema.GetFieldByName(fieldName)
-		if ok {
-			return NewFieldDefinition(
-				collectionField,
-				schemaField,
-			), true
-		}
+	collectionField, existsOnCollection := def.Description.GetFieldByName(fieldName)
+	schemaField, existsOnSchema := def.Schema.GetFieldByName(fieldName)
+
+	if existsOnCollection && existsOnSchema {
+		return NewFieldDefinition(
+			collectionField,
+			schemaField,
+		), true
+	} else if existsOnCollection && !existsOnSchema {
+		// If the field exists only on the collection, it is a local only field, for example the
+		// secondary side of a relation.
+		return NewLocalFieldDefinition(
+			collectionField,
+		), true
+	} else if !existsOnCollection && existsOnSchema {
```

Comment:
question: Can you explain how this will ever be the case? Even the embedded object doesn't seem to make sense to me here.

Reply:
We have this at the moment, e.g. related objects within a view (declared with `interface` instead of `type` in an SDL)